### PR TITLE
Stg/ashu/bck_localization

### PR DESCRIPTION
**Pull request by ashdude1401**
stg/ashu/Localization setup and translation addition in message properties

### CODE REVIEW

File: src/main/java/com/darzee/shankh/service/OrderService.java [hunk 1]
```diff
@@ -305,7 +362,7 @@ public ResponseEntity recieveOrderPayment(Long orderId, Long boutiqueId, Recieve
         Double pendingAmount = orderAmountDAO.getTotalAmount() - orderAmountDAO.getAmountRecieved();
         Double amountRecieved = request.getAmount();
         if (amountRecieved > pendingAmount) {
-            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Amount recieved is greater than pending order amount. Amount Received : " + amountRecieved + " Pending Amount : " + pendingAmount);
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST,localisationService.translate("Amount recieved is greater than pending order amount. Amount Received : ")  + amountRecieved +localisationService.translate(" Pending Amount : ")  + pendingAmount);
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/service/OrderService.java:365]: same as above

File: src/main/java/com/darzee/shankh/service/OrderService.java [hunk 2]
```diff
@@ -350,7 +407,7 @@ public ResponseEntity<InvoiceDetailResponse> getInvoiceDetail(Long orderId, Long
         try {
             OrderDAO orderDAO = findOrder(orderId, boutiqueId);
             if (orderDAO == null) {
-                return new ResponseEntity("Order not found", HttpStatus.NOT_FOUND);
+                return new ResponseEntity(localisationService.translate("Order not found"), HttpStatus.NOT_FOUND);
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/service/OrderService.java:410]: same as above

File: src/main/java/com/darzee/shankh/service/OrderService.java [hunk 3]
```diff
@@ -182,7 +218,28 @@ public ResponseEntity getOrderDetails(Long orderId, Long boutiqueId) throws Exce
         }
         Integer paymentMode = getOrderPaymentMode(order.getId());
         String message = "Details fetched succesfully";
-        OrderDetailResponse response = new OrderDetailResponse(customer, order, orderAmount, orderItemDetailsList, paymentMode, message);
+        OrderDetailResponse response = new OrderDetailResponse(customer, order, orderAmount, orderItemDetailsList, paymentMode,localisationService.translate(message));
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/service/OrderService.java:221]: translations should be handled inside OrderDetailResponse  constructor

File: src/main/java/com/darzee/shankh/service/StitchOptionService.java [hunk 4]
```diff
@@ -111,14 +114,38 @@ public GetStitchOptionsResponse getStitchOptions(Integer outfitId) {
         if (outfitType == null) {
             throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Invalid Outfit Id");
         }
-        List<StitchOptionsDAO> stitchOptions = mapper.stitchOptionListToStitchOptionDAOList(stitchOptionsRepo.findAllByOutfitTypeAndIsValidOrderByPriority(outfitType, Boolean.TRUE));
+        List<StitchOptionsDAO> stitchOptions =  mapper.stitchOptionListToStitchOptionDAOList(stitchOptionsRepo.findAllByOutfitTypeAndIsValidOrderByPriority(outfitType, Boolean.TRUE));
         GetStitchOptionsResponse stitchOptionsResponse = new GetStitchOptionsResponse(stitchOptions);
+        stitchOptionsResponse.getResponse().forEach(groupedStitchOptionDetails -> {
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/service/StitchOptionService.java:119]: handle translations inside the constructor

File: src/main/java/com/darzee/shankh/service/StitchOptionService.java [hunk 5]
```diff
@@ -111,14 +114,38 @@ public GetStitchOptionsResponse getStitchOptions(Integer outfitId) {
...
+            if (groupedStitchOptionDetails != null && groupedStitchOptionDetails.getStitchOptions() != null) {
+                groupedStitchOptionDetails.getStitchOptions().forEach(stitchOption -> {
+                    if (stitchOption != null) {
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/service/StitchOptionService.java:122]: instead of putting so many null checks here, do the null handling inside translate method

File: src/main/java/com/darzee/shankh/service/StitchOptionService.java [hunk 6]
```diff
@@ -111,14 +114,38 @@ public GetStitchOptionsResponse getStitchOptions(Integer outfitId) {
...
+            }
+        });
         return stitchOptionsResponse;
     }
 
     public ResponseEntity getOrderItemStitchOptionsResponse(Long orderItemId) {
-        Map<String, List<OrderStitchOptionDetail>> groupedOrderStitchOptionDetail = getOrderItemStitchOptions(orderItemId);
+        Map<String, List<OrderStitchOptionDetail>> groupedOrderStitchOptionDetail = getOrderItemStitchOptions(orderItemId);    
         GetOrderStitchOptionResponse response = new GetOrderStitchOptionResponse(groupedOrderStitchOptionDetail);
+        response.getResponse().forEach(
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/service/StitchOptionService.java:141]: inside constructor

File: src/main/java/com/darzee/shankh/service/LocalisationService.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+@Service
+public class LocalisationService {
+    private final MessageSource messageSource;
+
+    public LocalisationService(MessageSource messageSource) {
+        this.messageSource = messageSource;
+    }
+
+    public String translate(String englishText) {
+        String key = englishText.toLowerCase().replace(" ", ".");
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/service/LocalisationService.java:18]: add null check for englishText

File: src/main/java/com/darzee/shankh/response/GetStitchOptionsResponse.java [hunk 8]
```diff
@@ -12,26 +12,36 @@
...
+
+            GroupedStitchOptionDetails topStitchOptionDetails = new GroupedStitchOptionDetails(
+               OutfitSide.TOP.getView() , stitchOptionDetails);
             response.add(topStitchOptionDetails);
         }
         if(groupedByOutfitSide.containsKey(OutfitSide.BOTTOM)) {
-            List<StitchOptionDetail> stitchOptionDetails = groupedByOutfitSide.get(OutfitSide.BOTTOM).stream().map(stitchOptionsDAO ->
-                    new StitchOptionDetail(stitchOptionsDAO)).collect(Collectors.toList());
-            GroupedStitchOptionDetails bottomStitchOptionDetails = new GroupedStitchOptionDetails(OutfitSide.BOTTOM.getView(), stitchOptionDetails);
+            List<StitchOptionDetail> stitchOptionDetails = groupedByOutfitSide.get(OutfitSide.BOTTOM).stream().map(stitchOptionsDAO ->{
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/response/GetStitchOptionsResponse.java:39]: inside constructor

File: src/main/java/com/darzee/shankh/controller/OrderItemController.java [hunk 9]
```diff
@@ -73,7 +80,7 @@ public ResponseEntity<CreateStitchResponse> createStitchOptions(@Valid @RequestB
     public ResponseEntity<CreateStitchResponse> updateStitchOptions(@PathVariable("id") Long orderItemId,
                                                                     @Valid @RequestBody CreateStitchOptionRequest request) {
         StitchSummary stitchSummary = stitchOptionService.updateStitchOptions(orderItemId, request);
-        CreateStitchResponse createStitchResponse = new CreateStitchResponse("Stitch options created successfully",
+        CreateStitchResponse createStitchResponse = new CreateStitchResponse(localisationService.translate("Stitch options created successfully"),
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/controller/OrderItemController.java:83]: for this and trailing success messages, create a constants file, call it SuccessMessages. java, add these message strings as static messages there

class SuccessMessages {
    public static String STITCH_OPTIONS_SUCCESS = "whatever_messgae";
    
    public static String getTranslatedMessage(String message) {
        return localisationService.translate(message);
    }
}

do the same for all success messages

File: src/main/java/com/darzee/shankh/controller/OrderItemController.java [hunk 10]
```diff
@@ -50,6 +53,10 @@ public ResponseEntity<CreateOrderResponse> createOrderItem(@Valid @RequestBody O
     public ResponseEntity<OrderSummary> updateOrderItem(@PathVariable("id") Long orderItemId,
                                                         @Valid @RequestBody OrderItemDetailRequest orderItemDetails) throws Exception {
         OrderSummary orderSummary = orderOrderItemCommonService.updateOrderItem(orderItemId, orderItemDetails);
+    //     orderSummary.getOrderItemSummaryList().forEach(orderItemSummary -> {
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/controller/OrderItemController.java:56]: why is this removed

File: src/main/java/com/darzee/shankh/config/AppConfig.java [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+import org.springframework.web.servlet.i18n.AcceptHeaderLocaleResolver;
+import org.springframework.web.servlet.i18n.LocaleChangeInterceptor;
+
+@Configuration
+public class AppConfig implements WebMvcConfigurer {
+
+    @Bean
+    public LocaleResolver localeResolver() {
+        List<Locale> supportedLocales = Arrays.asList(
+                new Locale("en"),
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/config/AppConfig.java:24]: please use Languages enum class here. we should have a single source of truth

File: src/main/java/com/darzee/shankh/controller/OrderItemController.java (hunk 10 above)
Comment (ashdude1401) [reply src/main/java/com/darzee/shankh/controller/OrderItemController.java:56]: To translate outfit Alias , I had added this .But as now we do not need alias translation , that's why I commented out .If updated pr , I will remove the commented code .

File: src/main/java/com/darzee/shankh/service/LocalisationService.java (hunk 7 above)
Comment (ashdude1401) [reply src/main/java/com/darzee/shankh/service/LocalisationService.java:18]: Done Raghav

Comment (ashdude1401) [reply src/main/java/com/darzee/shankh/service/LocalisationService.java:18]: Can do that Raghav , we need to have some non null

File: src/main/java/com/darzee/shankh/service/StitchOptionService.java (hunk 5 above)
Comment (ashdude1401) [reply src/main/java/com/darzee/shankh/service/StitchOptionService.java:122]: Raghav I have added null check here because in stage db , for some order they are null and calling get and set function give null pointer error . These can't be handled inside tranlate function as before sending argument to the translate function we have to get the value using .getMethode().

File: src/main/java/com/darzee/shankh/config/AppConfig.java [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+import org.springframework.web.servlet.LocaleResolver;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+import org.springframework.web.servlet.i18n.AcceptHeaderLocaleResolver;
+import org.springframework.web.servlet.i18n.LocaleChangeInterceptor;
+
+@Configuration
+public class AppConfig implements WebMvcConfigurer {
+
+    public enum Language {
```
[the remark this answers is not in the document]
Comment (ashdude1401) [reply src/main/java/com/darzee/shankh/config/AppConfig.java:22]: done raghav

File: src/main/java/com/darzee/shankh/controller/OrderItemController.java (hunk 9 above)
Comment (ashdude1401) [reply src/main/java/com/darzee/shankh/controller/OrderItemController.java:79]: done Raghav

File: src/main/java/com/darzee/shankh/service/LocalisationService.java (hunk 7 above)
Comment (ashdude1401) [reply src/main/java/com/darzee/shankh/service/LocalisationService.java:19]: done Raghav

File: src/main/java/com/darzee/shankh/service/OrderService.java (hunk 2 above)
Comment (ashdude1401) [reply src/main/java/com/darzee/shankh/service/OrderService.java:409]: done

File: src/main/java/com/darzee/shankh/service/OrderService.java (hunk 3 above)
Comment (ashdude1401) [reply src/main/java/com/darzee/shankh/service/OrderService.java:221]: done

File: src/main/java/com/darzee/shankh/service/OrderService.java (hunk 1 above)
Comment (ashdude1401) [reply src/main/java/com/darzee/shankh/service/OrderService.java:364]: done

File: src/main/java/com/darzee/shankh/service/StitchOptionService.java (hunk 4 above)
Comment (ashdude1401) [reply src/main/java/com/darzee/shankh/service/StitchOptionService.java:119]: done

File: src/main/java/com/darzee/shankh/service/CustomerService.java [hunk 13]
```diff
@@ -106,8 +118,9 @@ public ResponseEntity createCustomer(CreateCustomerRequest request) {
         if (existingCustomer != null) {
             String customerName = existingCustomer.constructName();
             String gender = existingCustomer.getGender() != null ? existingCustomer.getGender().getString() : null;
+            String errorMessage = errorMessageTranslator.getTranslatedMessage(ErrorMessages.CUSTOMER_ALREADY_REGISTERED_ERROR_MSG);
             response = new CreateCustomerResponse(customerName, existingCustomer.getCountryCode(), existingCustomer.getPhoneNumber(),
-                    "", existingCustomer.getId(), gender, "Customer already registered");
+                    "", existingCustomer.getId(), gender, errorMessage);
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/service/CustomerService.java:123]: this is not an errorMessage cuz we're giving 200OK here

File: src/main/java/com/darzee/shankh/config/AppConfig.java (hunk 12 above)
[the remark this answers is not in the document]
Comment (raghav2499) [reply src/main/java/com/darzee/shankh/config/AppConfig.java:22]: we dont need to duplicate the code here. please use the eum directly

File: src/main/java/com/darzee/shankh/constants/ErrorMessages.java [hunk 14]
```diff
@@ -1,6 +1,54 @@
 package com.darzee.shankh.constants;
 
+import org.bouncycastle.mail.smime.handlers.pkcs7_mime;
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/constants/ErrorMessages.java:3]: why this import?

File: src/main/java/com/darzee/shankh/dao/LocalizationDAO.java [hunk 15]
```diff
@@ -0,0 +1,11 @@
+package com.darzee.shankh.dao;
+
+import java.time.LocalDateTime;
+
+public class LocalizationDAO {
```
[the remark this answers is not in the document]
Comment (raghav2499) [reply src/main/java/com/darzee/shankh/dao/LocalizationDAO.java:5]: if not required please remove this

File: src/main/java/com/darzee/shankh/service/BucketService.java [hunk 16]
```diff
@@ -11,10 +12,12 @@
 import com.darzee.shankh.response.DownloadImageResponse;
 import com.darzee.shankh.response.FileDetail;
 import com.darzee.shankh.response.UploadMultipleFileResponse;
+import com.darzee.shankh.service.translator.ErrorMessageTranslator;
 import com.darzee.shankh.utils.CommonUtils;
 import com.darzee.shankh.utils.s3utils.FileUtil;
 import org.apache.commons.lang3.tuple.ImmutablePair;
 import org.apache.commons.lang3.tuple.Pair;
+import org.apache.kafka.common.protocol.types.Field.Str;
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/service/BucketService.java:20]: why do we neeed this?

File: src/main/java/com/darzee/shankh/service/MeasurementService.java [hunk 17]
```diff
@@ -69,25 +78,43 @@ public class MeasurementService {
...
-                                                        Long orderItemId, Integer outfitTypeIndex,
-                                                        String scale, Boolean nonEmptyValuesOnly) throws Exception {
-        OverallMeasurementDetails measurementDetails = getMeasurementDetails(customerId, measurementRevisionId, orderItemId, outfitTypeIndex, scale, nonEmptyValuesOnly);
-        measurementDetails.setMessage(getMeasurementDetailsMessage(measurementDetails));
+            Long orderItemId, Integer outfitTypeIndex,
+            String scale, Boolean nonEmptyValuesOnly) throws Exception {
+        OverallMeasurementDetails measurementDetails = getMeasurementDetails(customerId, measurementRevisionId,
+                orderItemId, outfitTypeIndex, scale, nonEmptyValuesOnly);
+
+        final String message = getMeasurementDetailsMessage(measurementDetails);
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/service/MeasurementService.java:102]: why is this variable final?

File: src/main/java/com/darzee/shankh/service/OrderItemService.java [hunk 18]
```diff
@@ -320,7 +343,8 @@ private List<FileDetail> getAudioDetails(List<String> audioFileRefIds) {
 
     private void validateGetOrderItemRequest(Long boutiqueId, Long orderId) {
         if (boutiqueId == null && orderId == null) {
-            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Either Order ID or Boutique ID is mandatory");
+            final String errorMessage = errorMessageTranslator.getTranslatedMessage(ErrorMessages.INVALID_ORDER_ID_ERROR);
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/service/OrderItemService.java:346]: why is this final?

File: src/main/java/com/darzee/shankh/service/OrderService.java [hunk 19]
```diff
@@ -477,8 +500,8 @@ public OrderDAO updateOrderPostItemUpdation(OrderDAO orderDAO, Double refundAmou
         if (orderAmountDelta != 0) {
             if (refundAmount > 0) {
                 if (refundAmount > -orderAmountDelta) {
-                    throw new RuntimeException("Amount refund could not be greater than order amount change. " +
-                            "Order Amount changed by " + (-orderAmountDelta) + " and we cannot refund " + refundAmount);
+                    String errorMessage = errorMessageTranslator.getTranslatedMessage(ErrorMessages.REFUND_GREATER_THAN_ORDER_CHANGE)+ errorMessageTranslator.getTranslatedMessage(ErrorMessages.ORDER_AMOUNT_CHANGED )+(-orderAmountDelta)+errorMessageTranslator.getTranslatedMessage(ErrorMessages.REFUND_NOT_ALLOWED)+refundAmount;
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/service/OrderService.java:503]: instead of breaking, use placeholders here:

in message properties file
refund.message=Amount refund could not be greater than order amount change. Order Amount changed by {0} and we cannot refund {1}

in error messages
    public String getRefundMessage(double orderAmountDelta, double refundAmount, Locale locale) {
        Object[] args = { -orderAmountDelta, refundAmount };
        return messageSource.getMessage("refund.message", args, locale);
    }
    
    
    
    in OrderService
    translate(getRefundMessage(orderAmountDelta, refundAmount));

File: src/main/java/com/darzee/shankh/service/translator/GetOrderResponseTranslator.java [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+                //            orderDetailResponse.setMessage(localisationService.translate(orderDetailResponse.getMessage()));
+                            if(orderDetailResponse.getOrderItemDetails()!=null){
+                                orderDetailResponse.getOrderItemDetails().forEach(orderItemDetails -> {
+                                    if(orderItemDetails.getOutfitAlias()!=null) {
+                                        // orderItemDetails.setOutfitAlias(localisationService.translate(orderItemDetails.getOutfitAlias()));
+                                    }
+                                });
+                            }
+                            orderDetailResponse.getOrderItemDetails().forEach(orderItemDetails -> {
+                                if(orderItemDetails.getOrderItemStitchOptions()!=null){
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/service/translator/GetOrderResponseTranslator.java:35]: dont duplicate, use StitchOptionsTranslator

File: src/main/java/com/darzee/shankh/service/translator/GetOrderResponseTranslator.java [hunk 21]
```diff
@@ -0,0 +1,57 @@
...
+                                    orderItemDetails.getOrderItemStitchOptions().forEach((key, value) -> {
+                                                List<OrderStitchOptionDetail> orderStitchOptionDetails = value;
+                                                orderStitchOptionDetails.forEach(orderStitchOptionDetail -> {
+                                                    orderStitchOptionDetail.setLabel(localisationService.translate(orderStitchOptionDetail.getLabel()));                 orderStitchOptionDetail.setValue(localisationService.translate(orderStitchOptionDetail.getValue()));
+                                                });
+                                            }
+                                    );
+                                }
+                               if(orderItemDetails.getMeasurementDetails()!=null&&orderItemDetails.getMeasurementDetails().getInnerMeasurementDetails()!=null) {
+                                   orderItemDetails.getMeasurementDetails().getInnerMeasurementDetails().forEach(measurementDetails -> {
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/service/translator/GetOrderResponseTranslator.java:45]: use InnerMeasuremwentTranslator

File: src/main/java/com/darzee/shankh/service/translator/MeasurementDetailsTranslator.java [hunk 22]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.List;
+
+@Service
+public class MeasurementDetailsTranslator {
+
+    @Autowired
+    private LocalisationService localisationService;
+
+    public List<InnerMeasurementDetails> translate(List<InnerMeasurementDetails> measurementDetails) {
+        for(InnerMeasurementDetails measurementDetail : measurementDetails) {
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/service/translator/MeasurementDetailsTranslator.java:18]: use InnerMeasurementTranslator

File: src/main/java/com/darzee/shankh/service/translator/OrderDetailResponseTranslator.java [hunk 23]
```diff
@@ -0,0 +1,45 @@
...
+    private LocalisationService localisationService;
+
+    public String getTranslatedMessage(String message) {
+        return message;
+    }
+
+    public OrderDetailResponse getTransatedOrderDetailResponseTranslator(OrderDetailResponse response ){    
+        response.getOrderItemDetails().forEach(orderItemDetails -> {
+            // orderItemDetails.setOutfitAlias(localisationService.translate(orderItemDetails.getOutfitAlias()));
+            if(orderItemDetails.getOrderItemStitchOptions()!=null){
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/service/translator/OrderDetailResponseTranslator.java:25]: dont duplicate

File: src/main/java/com/darzee/shankh/service/translator/OrderDetailResponseTranslator.java [hunk 24]
```diff
@@ -0,0 +1,45 @@
...
+                orderItemDetails.getOrderItemStitchOptions().forEach((key, value) -> {
+                            List<OrderStitchOptionDetail> orderStitchOptionDetails = value;
+                            orderStitchOptionDetails.forEach(orderStitchOptionDetail -> {
+                                orderStitchOptionDetail.setLabel(localisationService.translate(orderStitchOptionDetail.getLabel()));                 orderStitchOptionDetail.setValue(localisationService.translate(orderStitchOptionDetail.getValue()));
+                            });
+                        }
+                );
+            }
+            if(orderItemDetails.getMeasurementDetails()!=null&&orderItemDetails.getMeasurementDetails().getInnerMeasurementDetails()!=null) {
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/service/translator/OrderDetailResponseTranslator.java:34]: dont duplicate

File: src/main/java/com/darzee/shankh/service/translator/StitchOptionResponseTranslator.java [hunk 25]
```diff
@@ -0,0 +1,39 @@
...
+    @Autowired
+    private LocalisationService localisationService;
+    public String getTranslatedMessage(String message) {    
+        return message;
+    }
+
+    public GetStitchOptionsResponse getTranslatedStitchOptionDetailList(GetStitchOptionsResponse stitchOptionsResponse) {
+        stitchOptionsResponse.getResponse().forEach(groupedStitchOptionDetails -> {
+            if (groupedStitchOptionDetails != null && groupedStitchOptionDetails.getStitchOptions() != null) {
+                groupedStitchOptionDetails.getStitchOptions().forEach(stitchOption -> {
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/service/translator/StitchOptionResponseTranslator.java:23]: use OrderStitchOptionsTranslator

File: src/main/java/com/darzee/shankh/config/AppConfig.java (hunk 12 above)
[the remark this answers is not in the document]
Comment (ashdude1401) [reply src/main/java/com/darzee/shankh/config/AppConfig.java:22]: Yes, we can directly use and set the locale from the Accept-Language header. However, it's important to declare all the supported languages for a few reasons:

Fallback mechanism: If the Accept-Language header is not present or if it's empty, your application needs to know which locale to use as a default. This is typically the first locale in our list of supported locales.

Invalid locales: The Accept-Language header might contain a locale that your application doesn't support. By declaring all the supported locales, you can check if the requested locale is supported and fall back to the default locale if it's not.

Locale matching: The Accept-Language header can contain multiple locales, each with a quality value that indicates the user's preference. The Locale.lookup method uses these quality values to find the best match among your supported locales.

In the provided code, the resolveLocale method first checks if the Accept-Language header is present and not empty. If it is, it uses Locale.lookup to find the best match among the supported locales. If the header is not present or empty, it falls back to the default locale.

[the remark this answers is not in the document]
Comment (ashdude1401) [reply src/main/java/com/darzee/shankh/config/AppConfig.java:22]: This is what I got ,when I did the search on whether we should define all the supported locale or not.

File: src/main/java/com/darzee/shankh/service/MeasurementService.java (hunk 17 above)
Comment (ashdude1401) [reply src/main/java/com/darzee/shankh/service/MeasurementService.java:102]: by mistake

File: src/main/java/com/darzee/shankh/service/OrderOrderItemCommonService.java [hunk 26]
```diff
@@ -190,25 +193,19 @@ private List<OrderItemDAO> collateAllItems(List<OrderItemDAO> allItems, List<Ord
...
-            OrderDAO orderDAO = orderItemDAO.getOrder();
-            Long orderNo = Optional.ofNullable(orderDAO.getBoutiqueOrderId()).orElse(orderDAO.getId());
-            try {
-                String url = generateItemDetailPdf(orderItemDAO, orderDAO.getCustomerId(), orderDAO.getBoutiqueId(),
-                        orderDAO.getBoutique().getName(), orderNo, language);
-                return url;
-            } catch(Exception e) {
-                throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "Error occurred while generating item PDF");
-            }
+        if(item.isPresent()){OrderItemDAO orderItemDAO = mapper.orderItemToOrderItemDAO(item.get(), new CycleAvoidingMappingContext());OrderDAO orderDAO = orderItemDAO.getOrder();
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/service/OrderOrderItemCommonService.java:198]: check this

File: src/main/java/com/darzee/shankh/service/translator/OrderStitchOptionsTranslator.java [hunk 27]
```diff
@@ -0,0 +1,55 @@
...
+                stitchOption.setValue(localisationService.translate(stitchOption.getValue())); 
+                        }
+                ) 
+        );
+        return response;
+    }
+
+    public List<StitchOptionDetail> translateStichOptionList(List<StitchOptionDetail> stitchOptionDetails ){
+
+        stitchOptionDetails.forEach(stitchOption -> {
```
Comment (raghav2499) [inline src/main/java/com/darzee/shankh/service/translator/OrderStitchOptionsTranslator.java:40]: remove duplicate

Comment (ashdude1401) [reply src/main/java/com/darzee/shankh/service/translator/OrderStitchOptionsTranslator.java:40]: Raghav I have separated the order Details Stitch option and stitch option translation separatly .

File: src/main/java/com/darzee/shankh/service/translator/GetOrderResponseTranslator.java (hunk 20 above)
Comment (ashdude1401) [reply src/main/java/com/darzee/shankh/service/translator/GetOrderResponseTranslator.java:35]: done raghav

File: src/main/java/com/darzee/shankh/service/OrderOrderItemCommonService.java (hunk 26 above)
Comment (ashdude1401) [reply src/main/java/com/darzee/shankh/service/OrderOrderItemCommonService.java:198]: okk

Comment (ashdude1401) [reply src/main/java/com/darzee/shankh/service/OrderOrderItemCommonService.java:198]: you have added this code .

File: src/main/java/com/darzee/shankh/service/OutfitService.java [hunk 28]
```diff
@@ -36,6 +41,7 @@ public ResponseEntity getOutfitDetails(String gender) throws Exception {
             outfitDetails.setStitchOptionsExist(stitchOptionsExist);
             outfitDetailsList.add(outfitDetails);
         }
+        outfitDetailsList =outfitTranslator.getTranslatedOutfitDetailsList(outfitDetailsList) ;
```
[the remark this answers is not in the document]
Comment (ashdude1401) [reply src/main/java/com/darzee/shankh/service/OutfitService.java:44]: It indentation is  correct in my local

File: src/main/java/com/darzee/shankh/service/translator/StitchOptionResponseTranslator.java (hunk 25 above)
Comment (ashdude1401) [reply src/main/java/com/darzee/shankh/service/translator/StitchOptionResponseTranslator.java:23]: Raghav as disscused we have different translator for StitchOptionResponseTranslator
